src/BackgroundTasks/Persistence.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: otruffer
+ * Date: 05.05.17
+ * Time: 10:14
+ */
+namespace ILIAS\BackgroundTasks;
+
+use ILIAS\BackgroundTasks\Implementation\Bucket\State;
+
+interface Persistence {
```

Comment:
Yeah, I like this pattern. For future reference: I think this is the "Repository Pattern". It helps testability a lot, imo, when contrasted with "Active Record".